sympy/physics/mechanics/body.py

```diff
@@ -214,3 +214,64 @@ def apply_torque(self, vec):
         if not isinstance(vec, Vector):
             raise TypeError("A Vector must be supplied to add torque.")
         self.loads.append((self.frame, vec))
+
+    def vel(self, body=None):
```

Comment:
The mass center does not move wrt to its body's frame, so you always need another body or frame here. I think you should accept a Body or a ReferenceFrame. Also a body can have an infinite number of points, so there are an infinite number of velocities. This should probably be named `masscenter_vel()` to be explicit about that.

Reply:
I'm trying to think whether we will every store other points on in a Body object other than the mass center. If we wouldn't then sticking with `vel()` as the name is ok. I'll think a bit about it.

Reply:
> I'm trying to think whether we will every store other points on in a Body object other than the mass center. If we wouldn't then sticking with `vel()` as the name is ok. I'll think a bit about it.

Current implementation stores masscenter only.

Reply:
I know that. But I'm trying to imagine the future. A body has an infinite number of points and some of those are important for any giving multi-body model. The first that comes to mind is locations that forces are applied on the body. But that only makes sense if we store forces in bodies.

Reply:
`Body.loads` does store point, force tuple.

Reply:
Ah, so there is already a `.loads` attribute. That's interesting. More thought needs to go into this, especially since forces/torques have to ensure equal and opposite behavior.

Reply:
@moorepants Yes, I think it would make applying forces and torques a bit easier.

```
class Forces:
    pass

F = Force(Body1, Body2)
```

Forces acting on Body1 can be +ve, and equal and opposite on Body 2, and can append the `.loads` and `.torque` of respective bodies. There is just one things that is tough is , to apply force on point other than masscenter, as we may need to take multiple parameters in `Force` for each body something like-
`F = Forces(Body1, Body1_point=None, Body2, Body2_point=None)` 
where `Body_point` is the point on each body on which force is applied.

Now problem 2 is gravity, as it should be wrt Newtonian frame, so shal we do something like-
`F = Forces(body1, body1_point=None, body2, body2_point=None, gravity=Frame/Body/None)`
If gravity is None then gravity is not applied, else a frame/Body should be supplied in which gravity is applied.

Reply:
I don't think we need to special case a constant gravitational field. You'd simple do `Force(Body1, Body1, g*Body1.x)` where Body1 is the newtonian frame. You have to do it for all bodies, but we can think of ways to ease things if you have lots of bodies to apply it to.

We should open a new issue or discussion for this topic, to try to work out the design.

Reply:
Once again, have some example problems that show how the api will be used on a real problem is essential.